### PR TITLE
Adapt typescript service examples to new spec

### PR DESCRIPTION
**Pull request by achim-k**
### Public-Facing Changes

Adapt typescript service examples to new spec

### Description
Uses the latest `advertiseService` spec update. Got slightly more complex but mainly due to an additional service endpoint being added.

### CODE REVIEW

File: typescript/ws-protocol-examples/package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "@foxglove/schemas": "^1.6.2",
     "@foxglove/wasm-lz4": "^1.0.2",
     "@foxglove/wasm-zstd": "^1.0.1",
-    "@foxglove/ws-protocol": "0.7.1",
+    "@foxglove/ws-protocol": "0.7.2",
```
Comment (jtbandes) [inline typescript/ws-protocol-examples/package.json:70]: is there no yarn.lock update from this? 🤔

File: typescript/ws-protocol-examples/src/examples/service-client.ts [hunk 2]
```diff
@@ -20,25 +24,27 @@ async function main(url: string, args: { encoding: "json" | "ros1" | "cdr" }) {
...
   client.on("advertiseServices", (services) => {
     const service = services.find((s) => /std_srvs(\/srv)?\/SetBool/.test(s.type));
     if (!service) {
       return;
     }
 
+    const msgEncoding = service.request?.encoding ?? fallbackMsgEncoding;
+    if (msgEncoding == undefined) {
+      throw new Error(
+        `Unable to call servicd ${service.name}: No supported message encoding found.`,
```
Comment (jtbandes) [inline typescript/ws-protocol-examples/src/examples/service-client.ts:40]:
```suggestion
        `Unable to call service ${service.name}: No supported message encoding found.`,
```

Comment (jtbandes) [reply typescript/ws-protocol-examples/src/examples/service-client.ts:40]: Maybe include the list of server `supportedEncodings` in the error message?

File: typescript/ws-protocol-examples/src/examples/service-server.ts [hunk 3]
```diff
@@ -25,13 +29,56 @@ async function main(): Promise<void> {
...
   };
-  const serviceId = server.addService(service);
+  const serviceDefJson: Omit<Service, "id"> = {
+    name: "/set_bool_json",
+    type: "std_srvs/SetBool",
+    request: {
+      encoding: "json",
+      schemaName: "SetBoolJsonRequest",
+      schemaEncoding: "jsonschema",
+      schema: `{
```
Comment (jtbandes) [inline typescript/ws-protocol-examples/src/examples/service-server.ts:55]: nit: you could do `schema: JSON.stringify({ ... })` and get some help from Prettier

**Review by jtbandes — APPROVED** [review]

File: typescript/ws-protocol-examples/package.json (hunk 1 above)
Comment (achim-k) [reply typescript/ws-protocol-examples/package.json:70]: Nope, was also surprised.

**Pull request merged**